tests/ecosystem/upgrade/test_upgrade.py

```diff
@@ -1,66 +1,58 @@
 import logging
 
+import pytest
+
 from ocs_ci.framework.testlib import ocs_upgrade
-from ocs_ci.framework import config
-from ocs_ci.ocs.cluster import CephCluster, CephHealthMonitor
-from ocs_ci.ocs.resources.storage_cluster import (
-    ocs_install_verification
-)
-from ocs_ci.utility.utils import TimeoutSampler
-from ocs_ci.ocs.ocs_upgrade import (
-    verify_image_versions,
-    OCSUpgrade,
-)
-from ocs_ci.ocs.exceptions import TimeoutException
+from ocs_ci.ocs.disruptive_operations import worker_node_shutdown
+from ocs_ci.ocs.ocs_upgrade import run_ocs_upgrade
+
 
 log = logging.getLogger(__name__)
 
 
+@pytest.fixture(autouse=True)
```

Comment:
this autouse fixture will be used for each function here in this test module so also for normal upgrade where we don't deal with restart nodes.  I am not sure if we want to have this finalizer called even for regular upgrade?

Also it's by default in functions cope, so it will be called after each tests if we have more tests defined in this module.  So I would not add it as autouse but use this fixture only in tests where it's needed.

Reply:
ok